pkg/agent/multicluster/stretched_networkpolicy_controller.go

```diff
@@ -0,0 +1,388 @@
+	}
+}
+
+func (s *StretchedNetworkPolicyController) processNextWorkItem() bool {
+	obj, quit := s.queue.Get()
+	if quit {
+		return false
+	}
+	defer s.queue.Done(obj)
+
+	if p, ok := obj.(*v1.Pod); !ok {
+		s.queue.Forget(obj)
+		klog.Errorf("Expected type 'pod' in work queue but got object", "object", obj)
+	} else if err := s.syncPodClassifierFlow(p); err == nil {
+		s.queue.Forget(p)
+	} else {
+		// Put the item back on the workqueue to handle any transient errors.
+		s.queue.AddRateLimited(p)
+		klog.ErrorS(err, "Error syncing podClassifierFlowUpdateItem, requeuing", "podName", p.Name, "podNS", p.Namespace)
+	}
+	return true
+}
+
+// syncPodClassifierFlow will update labelToPods and podToLabel and update
+// PodClassifierFlow with its LabelIdentity, if available, or UnknownLabelIdentity.
+func (s *StretchedNetworkPolicyController) syncPodClassifierFlow(pod *v1.Pod) error {
+	podRef := getPodReference(pod)
+	podNS, err := s.namespaceLister.Get(pod.Namespace)
+	if err != nil {
+		return err
+	}
+	normalizedLabel := getNormalizedLabel(podNS.Labels, pod.Labels, podNS.Name)
+	s.lock.Lock()
+	defer s.lock.Unlock()
+	oldNormalizedLabel, ok := s.podToLabel[podRef]
+	if ok && oldNormalizedLabel != normalizedLabel {
+		s.deleteLabelToPod(oldNormalizedLabel, pod)
+	}
+	if !ok || oldNormalizedLabel != normalizedLabel {
+		s.addLabelToPod(normalizedLabel, pod)
+		s.podToLabel[getPodReference(pod)] = normalizedLabel
+	}
+
+	if id, ok := s.labelIdentityCache[normalizedLabel]; ok {
+		return s.updatePodClassifierFlowWithLabelIdentity(pod, id)
+	}
+	return s.updatePodClassifierFlowWithLabelIdentity(pod, openflow.UnknownLabelIdentity)
+}
+
+// processPodUpdate handles Pod add events. It enqueues the Pod.
+func (s *StretchedNetworkPolicyController) processPodAdd(cur interface{}) {
+	curPod, curIsPod := cur.(*v1.Pod)
+	if !curIsPod {
+		klog.V(2).InfoS("StretchedNetworkPolicyController Pod AddFunc received unexpected object", "curObj", cur)
+		return
+	}
+	s.queue.Add(curPod)
+}
+
+// processPodUpdate handles Pod update events. It only enqueues the Pod if the
+// Labels of this Pod has been updated.
+func (s *StretchedNetworkPolicyController) processPodUpdate(old, cur interface{}) {
+	oldPod, oldIsPod := old.(*v1.Pod)
+	curPod, curIsPod := cur.(*v1.Pod)
+	if !oldIsPod || !curIsPod {
+		klog.V(2).InfoS("StretchedNetworkPolicyController Pod UpdateFunc received unexpected object", "oldObj", old, "curObj", cur)
+		return
+	}
+	if oldPod != nil && reflect.DeepEqual(oldPod.Labels, curPod.Labels) {
+		klog.V(2).InfoS("StretchedNetworkPolicyController Pod UpdateFunc received a Pod update event, "+
+			"but labels are the same. Skip it", "podName", curPod.Name, "podNamespace", curPod.Namespace)
+		return
+	}
+	s.queue.Add(curPod)
+}
+
+// processPodDelete handles Pod delete events. It deletes the Pod from the
+// labelToPods and podToLabel. After Pod is deleted, its classifier flow will also
+// be deleted. So no need to enqueue this Pod to update its classifier flow.
```

Comment:
Could you also mention which controller is responsibly for deleting classifier flow, for debuggabiltity purposes?

Reply:
Done.